doc/semantics.md

```diff
@@ -48,44 +60,80 @@ type they were created with. The following are examples:
 - 1 matches `Byte`, `Short`, `Int`, `Float`, `Double`
 - 128 (`> Byte.MaxValue`) matches `Short`, `Int`, `Float`, `Double`
 - 32768 (`> Short.MaxValue`) matches `Int`, `Float`, `Double`
+- 2147483647 matches `Int`, `Double` if strict-floats are enabled,
+  otherwise `Float` as well
 - 2147483648 (`> Int.MaxValue`) matches `Float`, `Double`
-- 1.2 matches `Float`, `Double`
+- 1.5 matches `Float`, `Double`
+- 1.4 matches `Double` only if strict-floats are enabled,
+  otherwise `Float` and `Double`
```

Comment:
It doesn't show in the short diff, but I added the following line right after this one:

> (unlike 1.5, the value 1.4 cannot be represented in a strict 32-bit `Float`)